regions/regions.go

```diff
@@ -44,14 +46,47 @@ func (rg Regions) RoundTrip(req *http.Request) (*http.Response, error) {
 	}
 	shardsRing, ok := rg.multiCluters[reqHost]
 	if ok {
+		req.Header.Add(storage.InternalHostHeader, reqHost)
+		req.Header.Add(storage.InternalBucketHeader, "")
 		return shardsRing.DoRequest(req)
 	}
+	host, bucketName := rg.findHostInDomainStyle(reqHost)
+	if host != "" && bucketName != "" {
+		shardsRing, _ = rg.multiCluters[host]
+		req.Header.Add(storage.InternalHostHeader, host)
+		req.Header.Add(storage.InternalBucketHeader, bucketName)
+		return shardsRing.DoRequest(req)
+
+	}
 	if rg.defaultRing != nil {
 		return rg.defaultRing.DoRequest(req)
 	}
 	return rg.getNoSuchDomainResponse(req), nil
 }
 
+func (rg Regions) findHostInDomainStyle(originalHost string) (currentHost, bucket string) {
```

Comment:
If we'll decide using func. `findHostInDomainStyle` I'll be recommending to move func. into `regions/utils` subpackage.